.github/files/gh-autorelease/files/autorelease.sh

```diff
@@ -2,32 +2,42 @@
 
 set -eo pipefail
 
+: "${GH_TOKEN:?Build argument needs to be set and non-empty.}"
 : "${GITHUB_REF:?Build argument needs to be set and non-empty.}"
 : "${GITHUB_SHA:?Build argument needs to be set and non-empty.}"
-: "${API_TOKEN_GITHUB:?Build argument needs to be set and non-empty.}"
-: "${GITHUB_API_URL:?Build argument needs to be set and non-empty.}"
-: "${GITHUB_REPOSITORY:?Build argument needs to be set and non-empty.}"
 
-## Determine tag
-if [[ ! "$GITHUB_REF" =~ ^refs/tags/v?[0-9]+(\.[0-9]+)+(-[a-z0-9._-]+)?$ ]]; then
-	echo "::error::Expected GITHUB_REF like \`refs/tags/v1.2.3\` or \`refs/tags/1.2.3\`, got \`$GITHUB_REF\`"
+if [[ ! -f composer.json ]]; then
+	echo '::error::No composer.json. Did it get excluded from the mirror?'
 	exit 1
 fi
-TAG="${GITHUB_REF#refs/tags/}"
 
-## Check for alphas
-if [[ "$TAG" =~ -(alpha|a\.[0-9]*[02468])$ ]]; then
-	echo "Not creating a release for alpha version $TAG"
-	exit 0
-fi
-echo "Creating release for $TAG"
+## Determine tag
+ROLLING_MODE=
+if [[ "$GITHUB_REF" =~ ^refs/tags/v?[0-9]+(\.[0-9]+)+(-[a-z0-9._-]+)?$ ]]; then
+	TAG="${GITHUB_REF#refs/tags/}"
 
-## Determine slug and title format.
-if [[ ! -f composer.json ]]; then
-	echo '::error::No composer.json. Did it get excluded from the mirror?'
+	## Check for alphas
+	if [[ "$TAG" =~ -(alpha|a\.[0-9]*[02468])$ ]]; then
+		echo "Not creating a release for alpha version $TAG"
+		exit 0
+	fi
+elif [[ "$GITHUB_REF" == "refs/heads/trunk" ]]; then
+	if ! jq -e '.extra.autorelease["rolling-release"]? // false' composer.json > /dev/null; then
+		echo "::notice::Skipping trunk release because autorelease rolling mode is not enabled."
+		exit 0
+	fi
+	ROLLING_MODE=true
+	CURRENT_VER=$( sed -nEe 's/^## \[?([^]]*)\]? - .*/\1/;T;p;q' CHANGELOG.md || true )
+	GIT_SUFFIX=$( git log -1 --format=%ct . )
```

Comment:
Since as far as semver is concerned the `+rolling.whatever` build component isn't a sortable part of the version string, we may as well just use the hash instead
```suggestion
	GIT_SUFFIX=$( git log -1 --format=%h . )
```

OTOH, if we have to go back to something more like `1.2.3-alpha.xxx`, switching back to the timestamp would probably be better to keep the ordering correct.

Reply:
Yeah, I agree, hashes could be used to look up the actual commit the release was created from.